### PR TITLE
[Microsoft.AwsConnector] Add s3AccessControlPolicy and remove emrClustersummary resource type in GA 2024-12-01 API version

### PR DESCRIPTION
**Pull request by PriyankaHariharan**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/awsconnector/S3AccessControlPolicy.Management/main.tsp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+@doc("Definition of Grantee")
+model Grantee {
+  @doc("<p>Screen name of the grantee.</p>")
+  displayName?: string;
+
+  @doc("<p>Email address of the grantee.</p> <note> <p>Using email addresses to specify a grantee is only supported in the following Amazon Web Services Regions: </p> <ul> <li> <p>US East (N. Virginia)</p> </li> <li> <p>US West (N. California)</p> </li> <li> <p> US West (Oregon)</p> </li> <li> <p> Asia Pacific (Singapore)</p> </li> <li> <p>Asia Pacific (Sydney)</p> </li> <li> <p>Asia Pacific (Tokyo)</p> </li> <li> <p>Europe (Ireland)</p> </li> <li> <p>South America (São Paulo)</p> </li> </ul> <p>For a list of all the Amazon S3 supported Regions and endpoints, see <a href='https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region'>Regions and Endpoints</a> in the Amazon Web Services General Reference.</p> </note>")
+  emailAddress?: string;
+
+  @doc("<p>The canonical user ID of the grantee.</p>")
+  id?: string;
```
Comment (mentat9) [inline specification/awsconnector/S3AccessControlPolicy.Management/main.tsp:152]: >id [](http://example.com/codeflow?start=2&length=2)

Don't reuse top-level ARM names within the property bag. Something like userId would work.

File: specification/awsconnector/S3AccessControlPolicy.Management/main.tsp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+  @doc("<p>Type of grantee</p>")
+  type?: TypeEnumValue;
```
Comment (mentat9) [inline specification/awsconnector/S3AccessControlPolicy.Management/main.tsp:155]: >type [](http://example.com/codeflow?start=2&length=4)

Same here, maybe granteeType?

**Comment by PriyankaHariharan** [comment]
Greetings ARM Reviewer @mentat9 ,

Thank you for reviewing this PR!

Please scan the below meeting minutes screen shot to get guidelines suggested by @rkmanda for reviewing this PR.

image
Related meeting minutes email attached as PDF: [Asset management - ARM API guidelines.pdf](https://github.com/Azure/azure-rest-api-specs-pr/files/15155958/Asset.management.-.ARM.API.guidelines.pdf)

Previous PR for this RP: https://github.com/Azure/azure-rest-api-specs-pr/pull/18073#issuecomment-2083860551

File: specification/awsconnector/S3AccessControlPolicy.Management/main.tsp (hunk 1 above)
Comment (PriyankaHariharan) [reply specification/awsconnector/S3AccessControlPolicy.Management/main.tsp:152]: Hi Chris, thank you for reviewing this PR, this is determined by AWS schema and is out of our control. I have added ARM review guidance for our RP in [this comment in this PR ](https://github.com/Azure/azure-rest-api-specs/pull/30965#issuecomment-2414800794)

File: specification/awsconnector/S3AccessControlPolicy.Management/main.tsp (hunk 2 above)
Comment (PriyankaHariharan) [reply specification/awsconnector/S3AccessControlPolicy.Management/main.tsp:155]: Hi Chris, thank you for reviewing this PR, this is determined by AWS schema and is out of our control. I have added ARM review guidance for our RP in https://github.com/Azure/azure-rest-api-specs/pull/30965#issuecomment-2414800794

File: specification/awsconnector/S3AccessControlPolicy.Management/main.tsp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+  @doc("<p>URI of the grantee group.</p>")
+  uri?: string;
```
[the remark this answers is not in the document]
Comment (PriyankaHariharan) [reply specification/awsconnector/S3AccessControlPolicy.Management/main.tsp:158]: Hi Chris, thank you for reviewing this PR, this is determined by AWS schema and is out of our control. I have added ARM review guidance for our RP in https://github.com/Azure/azure-rest-api-specs/pull/30965#issuecomment-2414800794

**Comment by mentat9** [comment]
Waiving feedback on the property bag and signing off for ARM based on the MultiCloudIntegration label.

**Review by ikanni — APPROVED** [review]

**Pull request merged**